packages/integrations/src/download-client/aria2/aria2-integration.ts

```diff
@@ -0,0 +1,151 @@
+import { fetchWithTrustedCertificatesAsync } from "@homarr/certificates/server";
+
+import type { DownloadClientJobsAndStatus } from "../../interfaces/downloads/download-client-data";
+import { DownloadClientIntegration } from "../../interfaces/downloads/download-client-integration";
+import type { DownloadClientItem } from "../../interfaces/downloads/download-client-items";
+import { CreateAria2, Aria2DownloadState } from './aria2-client';
+import path from "path"
+
+
+export class Aria2Integration extends DownloadClientIntegration {
+  public async getClientJobsAndStatusAsync(): Promise<DownloadClientJobsAndStatus> {
+    const client = await this.getClientAsync();
+    // ?: Should recursive retrive all items or just few first item?
+    const [actives, waitings, stoppeds, globalstats] = await Promise.all([client.tellActive(), client.tellWaiting(0, 1000), client.tellStopped(0, 1000), client.getGlobalStat()]);
+
+    const downloadings = [...actives, ...waitings, ...stoppeds];
+
+    return {
+      status: {
+        types: ["torrent", "http(s)"],
+        paused: false,
+        rates: {
+          up: Number(globalstats.uploadSpeed),
+          down: Number(globalstats.downloadSpeed),
+        }
+      },
+      items: downloadings.map((item, idx) => {
+        const totalSize = Number(item.totalLength);
+        const completedSize = Number(item.completedLength);
+        const progress = totalSize > 0 ? completedSize / totalSize : 0;
+        const itemName = item?.bittorrent?.info?.name || path.basename(item.files[0]?.path || "Unknown");
+
+        return {
+          id: item.gid,
+          index: idx,
+          name: itemName,
+          type: !!item.bittorrent ? "torrent" : "http(s)",
+          size: totalSize,
+          sent: Number(item.uploadLength),
+          downSpeed: Number(item.downloadSpeed),
+          upSpeed: Number(item.uploadSpeed),
+          time: this.calculateEta(completedSize, totalSize, Number(item.downloadSpeed)),
+          state: this.getState(item.status, !!item.bittorrent),
+          progress: progress,
+          category: [],
+        }
+      }),
+    } as DownloadClientJobsAndStatus
+  }
+  public async pauseQueueAsync(): Promise<void> {
+
+    const client = await this.getClientAsync();
+    await client.pauseAll();
+  }
+  public async pauseItemAsync(item: DownloadClientItem): Promise<void> {
+    const client = await this.getClientAsync();
+    switch (item.state) {
+      case "downloading":
+      case "leeching":
+        await client.pause(item.id);
+        return;
+      default:
+        return;
+    }
```

Comment:
This is bad practice - there is only one branch and AFAIK you won't get any performance advantage (is this correct  @Meierschlumpf )? Use a normal ``if`` statement instead.